File: Sources/FitUI/Components/FitPasswordRule.swift
import SwiftUI

// MARK: - FitPasswordRule
//
// Vertical list of password validation rules — 12×12 check icon
// (tertiary if unmet, teal if met) + 14pt rule text.
// See `docs/components.md` § FitPasswordRule.

public struct FitPasswordRuleItem: Identifiable, Hashable {
    public let id: UUID
    public let label: String
    public let isMet: Bool

    public init(id: UUID = UUID(), label: String, isMet: Bool) {
        self.id = id
        self.label = label
        self.isMet = isMet
    }
}

public struct FitPasswordRule: View {
    let rules: [FitPasswordRuleItem]

    @Environment(\.fitTheme) private var theme

    public init(rules: [FitPasswordRuleItem]) {
        self.rules = rules
    }

    public var body: some View {
        VStack(alignment: .leading, spacing: 6) {
            ForEach(rules) { rule in
                HStack(spacing: FitSpacing.sp2) {
                    Image(systemName: rule.isMet ? "checkmark" : "circle")
                        .font(.system(size: 10, weight: .bold))
                        .foregroundColor(rule.isMet ? FitColors.Teal.t500 : theme.textTertiary)
                        .frame(width: 12, height: 12)

                    Text(rule.label)
                        .font(.custom(FitFont.family, size: 14))
                        .foregroundColor(rule.isMet ? theme.textSecondary : theme.textTertiary)
                }
            }
        }
    }
}

File: android/src/main/kotlin/com/fit321/fitui/components/FitPasswordRule.kt
package com.fit321.fitui.components

import androidx.compose.foundation.layout.*
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.Check
import androidx.compose.material.icons.filled.Circle
import androidx.compose.material3.Icon
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.unit.dp
import com.fit321.fitui.theme.LocalFitTheme
import com.fit321.fitui.tokens.FitColors
import com.fit321.fitui.tokens.FitFont
import com.fit321.fitui.tokens.FitSpacing

// ============================================================================
// FitPasswordRule — vertical list of password validation rules.
// Mirrors Swift FitPasswordRule. See `docs/components.md` § FitPasswordRule.
// ============================================================================

data class FitPasswordRuleItem(val label: String, val isMet: Boolean)

@Composable
fun FitPasswordRule(
    rules: List<FitPasswordRuleItem>,
    modifier: Modifier = Modifier
) {
    val theme = LocalFitTheme.current
    Column(
        modifier = modifier,
        verticalArrangement = Arrangement.spacedBy(6.dp)
    ) {
        rules.forEach { rule ->
            Row(
                verticalAlignment = Alignment.CenterVertically,
                horizontalArrangement = Arrangement.spacedBy(FitSpacing.sp2)
            ) {
                Icon(
                    imageVector = if (rule.isMet) Icons.Default.Check else Icons.Default.Circle,
                    contentDescription = null,
                    tint = if (rule.isMet) FitColors.Teal.t500 else theme.textTertiary,
                    modifier = Modifier.size(12.dp)
                )
                Text(
                    rule.label,
                    style = FitFont.body2,
                    color = if (rule.isMet) theme.textSecondary else theme.textTertiary
                )
            }
        }
    }
}
